packages/react/src/contents/hooks/useCommercePages.ts

```diff
@@ -1,91 +1,87 @@
 import {
-  fetchCommercePages as fetchCommercePagesAction,
+  ContentTypeCode,
+  fetchCommercePages,
+  getContentByQuery,
   getContentError,
   getContents,
+  isContentFetched,
   isContentLoading,
-  resetContents as resetContentsAction,
   StoreState,
 } from '@farfetch/blackout-redux';
-import { useEffect } from 'react';
+import {
+  Strategy,
+  UseCommercePageOptions,
+} from './types/useCommercePages.types';
+import { useCallback, useEffect, useMemo } from 'react';
 import { useSelector } from 'react-redux';
 import useAction from '../../helpers/useAction';
-import type { CommercePagesParams } from '../types';
+import type { ComponentType } from '@farfetch/blackout-client';
 
-/**
- * Hook to return actions and selectors for Commerce page data. The action to fetch
- * content will be automatically called so there is no need to refetch.
- *
- * @param slug      - Slug of the commerce page to fetch for data.
- * @param params    - The target parameters that a commerce page is configured.
- * @param pageIndex - Number of the page to get, starting at 1. The default is 1.
- * @param pageSize  - Size of each page, as a number between 1 and 180. The default is 60.
- * @param strategy  - Strategy to get best ranking commerce page.
- *
- * @returns - Returns actions and selectors for commerce page data.
- */
-const useCommercePages = (
-  slug: string,
-  params: CommercePagesParams,
-  pageIndex?: number,
-  pageSize?: number,
-  strategy = 'default',
+const useCommercePages = <T = ComponentType[]>(
+  options: UseCommercePageOptions,
 ) => {
-  const queryCommerce = {
-    type: params.type,
-    id: params?.id,
-    gender: params?.gender,
-    brand: params?.brand,
-    category: params?.category,
-    priceType: params?.priceType,
-    sku: params?.sku,
-    pageSize,
-    pageIndex,
-  };
-  const queryContent = {
-    codes: slug,
-    contentTypeCode: 'commerce_pages',
-    pageIndex,
-    pageSize,
-  };
+  const {
+    enableAutoFetch = true,
+    fetchConfig,
+    strategy = Strategy.Default,
```

Comment:
From what I can see after analysing the code from both the `fetchCommercePagesFactory` action and `getCommercePages` client, it seems to me that if `strategy` is not passed, everything works. If that is the case, do we really need to apply a default value here for strategy?